go-controller/cmd/ovn-k8s-cni-overlay/app/helper_windows.go

```diff
@@ -0,0 +1,220 @@
+			logrus.Warningf("Failed to delete HNS endpoint: %q", err)
+		} else {
+			logrus.Infof("HNS endpoint successfully deleted: %q", endpointName)
+		}
+	} else {
+		logrus.Infof("No endpoint with name %v was found, error %v", endpointName, err)
+	}
+	// Return the error in case it failed, we don't want to leak any HNS Endpoints
+	return err
+}
+
+// ConfigureInterface sets up the container interface
+// Small note on this, the call to this function should be idempotent on Windows.
+// The fact that CNI add should be idempotent on Windows is stated here:
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/network/cni/cni_windows.go#L38
+func ConfigureInterface(args *skel.CmdArgs, namespace string, podName string, macAddress string, ipAddress string, gatewayIP string) ([]*current.Interface, error) {
+	if strings.HasPrefix(args.Netns, containerNetNS) || strings.Compare(args.Netns, noNameNetNS) != 0 {
+		// If it is a normal container from the pod, there is nothing to do.
+		// Also, if it is not an infra container, nothing to do in this case as well.
+		logrus.Infof("CNI called for normal container or infra container, nothing to do")
+		return []*current.Interface{}, nil
+	}
+
+	ipAddr, ipNet, err := net.ParseCIDR(ipAddress)
+	if err != nil {
+		return nil, err
+	}
+	ipMaskSize, _ := ipNet.Mask.Size()
+	endpointName := args.ContainerID
+
+	var hnsNetworkId string
+	hnsNetworkId, err = getHNSIdFromConfigOrByGatewayIP(gatewayIP)
+	if err != nil {
+		logrus.Infof("Error when detecting the HNS Network Id: %q", err)
+		return nil, err
+	}
+
+	// Ensure that the macAddress is given in xx:xx:xx:xx:xx:xx format
+	macAddressIpFormat := strings.Replace(macAddress, ":", "-", -1)
+
+	// Check if endpoint is created, otherwise create it.
+	// This is to make the call to add idempotent
+	var createdEndpoint *hcsshim.HNSEndpoint
+	createdEndpoint, err = hcsshim.GetHNSEndpointByName(endpointName)
+	if err != nil {
+		logrus.Infof("HNS endpoint %q does not exist", endpointName)
+		hnsEndpoint := &hcsshim.HNSEndpoint{
+			Name:           endpointName,
+			VirtualNetwork: hnsNetworkId,
+			IPAddress:      ipAddr,
+			MacAddress:     macAddressIpFormat,
+			PrefixLength:   uint8(ipMaskSize),
+		}
+		createdEndpoint, err = createHNSEndpoint(hnsEndpoint)
+		if err != nil {
+			logrus.Warningf("HNS endpoint failed to create, reason: %q", err)
+			return nil, err
+		}
+	} else {
+		logrus.Infof("HNS endpoint already exists with name: %q", endpointName)
+	}
+
+	err = containerHotAttachEndpoint(createdEndpoint, args.ContainerID)
+	if err != nil {
+		logrus.Warningf("Failed to hot attach HNS Endpoint %q to container %q, reason: %q", endpointName, args.ContainerID, err)
+		return nil, err
+	}
+
+	ifaceID := fmt.Sprintf("%s_%s", namespace, podName)
+	// "--may-exist"  is added to support the function idempotency
+	ovsArgs := []string{
+		"--may-exist", "add-port", "br-int", endpointName, "--", "set",
+		"interface", endpointName, "type=internal", "--", "set",
```

Comment:
Mind adding a TODO here. `br-int` can be set/get via `external_ids` if we want in the future. Also the `type=internal` has to be revisited later on when we have to deal with `Hyper-V containers`, please add a TODO for that as well.

Reply:
That change for ``br-int`` would require changing it in other places as well and adding it in the config file as integration_bridge. This wouldn't be the only change we would need to do if we want to have something else for ``br-int``, I think that it's good enough for the moment.
Gonna add a TODO for ``Hyper-V Containers``